### PR TITLE
Simplify modbus frequency unittest

### PR DESCRIPTION
**Pull request by agutenkunst**
Addresses http://build.ros.org/job/Mdev__pilz_robots__ubuntu_bionic_amd64/67/testReport/junit/(root)/PilzModbusClientTests/testSettingReadFrequency/

### CODE REVIEW

File: prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp [hunk 1]
```diff
@@ -439,29 +439,16 @@ class RegisterBuffer
...
  *
  *
  * Test Sequence:
- * - 1. Choose low read frequency (f1). The last received "modbus_read" message
- *      is stored in a buffer.
- * - 2. Choose "modbus_read" topic check frequency f2, with f2 slightly bigger than f1.
- * - 3. Start a loop (running exactly for \#f1 iterations), which checks with
- *      frequency f2 the last received "modbus_read" message.
- * - 4. Stop all loops and the PilzModbusClient.
+ * - 1. Choose a read frequency. Measure the number of register accesses within ~3 seconds.
```
Comment (martiniil) [inline prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp]: In order to check if a frequency can be changed, the whole test case should be performed twice with different read frequencies.

Comment (martiniil) [reply prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp]: Can we reduce to ~1 sec?

**Review by martiniil — CHANGES_REQUESTED** [review]

File: prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp [hunk 2]
```diff
@@ -491,40 +478,21 @@ TEST_F(PilzModbusClientTests, testSettingReadFrequency)
...
-  EXPECT_GE(buffer.get(), static_cast<uint16_t>(expected_read_frequency)) << "Frequency used by PilzModbusClient is probably too low";
+  // Time for measuring
+  ros::Duration(3).sleep();
 
+  auto final_value = buffer.get();
+  auto time_stop = ros::Time::now();
   executor.stop();
+
+  // Checks
+  EXPECT_NEAR(expected_read_frequency, (final_value - 1) / (time_stop - time_start).toSec(), 0.1 * expected_read_frequency);
```
Comment (martiniil) [inline prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp]: You could increase readability by splitting up this line, like e.g.
```
actual_read_frequence = ...
allowed_error = ...
```

File: prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp (hunk 1 above)
Comment (agutenkunst) [reply prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp]: Done. Done.

File: prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp [hunk 3]
```diff
@@ -439,93 +444,69 @@ class RegisterBuffer
...
- *      set properly and is probably to high.
- * - 4. Check that the number of received messages is greater than \#f1. If the
- *      condition does not hold, the read fequency is not set properly
- *      and is probably to low.
+ * - 1. The measured frequence is within a 10% tolerance
  */
 TEST_F(PilzModbusClientTests, testSettingReadFrequency)
 {
-  std::unique_ptr<PilzModbusClientMock> mock(new PilzModbusClientMock());
+  std::vector<double> checked_frequencies {30.0, 100.0, 1000.0};
```
Comment (martiniil) [inline prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp]: Isn't this a bit optimistic? I think we have to choose smaller frequencies. At least I got a failure:
```
/home/imartini/catkin_ws/src/pilz_robots/prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp:504: Failure
The difference between expected_read_frequency and actual_frequency is 634.70369824667364, which exceeds 0.1 * expected_read_frequency, where
expected_read_frequency evaluates to 1000,
actual_frequency evaluates to 365.2963017533263, and
0.1 * expected_read_frequency evaluates to 100.
```

**Review by martiniil — CHANGES_REQUESTED** [review]

File: prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp (hunk 3 above)
Comment (agutenkunst) [reply prbt_hardware_support/test/unittests/unittest_pilz_modbus_client.cpp]: Done.

**Review by martiniil — APPROVED** [review]

**Pull request merged**